pkg/engine/resolve/resolve_test.go

```diff
@@ -1514,6 +1514,106 @@ func TestResolver_ResolveNode(t *testing.T) {
 			},
 		}, Context{Context: context.Background()}, `{"id":1,"name":"Jens","pet":{"name":"Woofie"}}`
 	}))
+	t.Run("json object within a string", testFn(false, false, func(t *testing.T, ctrl *gomock.Controller) (node Node, ctx Context, expectedOutput string) {
+		return &Object{
+			Fetch: &SingleFetch{
+				BufferId:   0,
+				DataSource: FakeDataSource(`{"data":"{ \"hello\": \"world\" }"}`),
+			},
+			Nullable: false,
+			Fields: []*Field{
+				{
+					Name: []byte("data"),
+					Value: &String{
+						Path:                 []string{"data"},
+						Nullable:             true,
+						UnescapeResponseJson: true,
+						IsTypeName:           false,
+					},
+					Position: Position{
+						Line:   2,
+						Column: 3,
+					},
+					HasBuffer: true,
+				},
+			},
+		}, Context{Context: context.Background()}, `{"data":{ "hello": "world" }}`
+	}))
+	t.Run("json array within a string", testFn(false, false, func(t *testing.T, ctrl *gomock.Controller) (node Node, ctx Context, expectedOutput string) {
+		return &Object{
+			Fetch: &SingleFetch{
+				BufferId:   0,
+				DataSource: FakeDataSource(`{"data": "[1, 2, 3]"}`),
+			},
+			Nullable: false,
+			Fields: []*Field{
+				{
+					Name: []byte("data"),
+					Value: &String{
+						Path:                 []string{"data"},
+						Nullable:             true,
+						UnescapeResponseJson: true,
+						IsTypeName:           false,
+					},
+					Position: Position{
+						Line:   2,
+						Column: 3,
+					},
+					HasBuffer: true,
+				},
+			},
+		}, Context{Context: context.Background()}, `{"data":[1, 2, 3]}`
+	}))
+	t.Run("json boolean within a string", testFn(false, false, func(t *testing.T, ctrl *gomock.Controller) (node Node, ctx Context, expectedOutput string) {
+		return &Object{
+			Fetch: &SingleFetch{
+				BufferId:   0,
+				DataSource: FakeDataSource(`{"data": "true"}`),
```

Comment:
hmm, looks like an edge case

upstream is returning a string `"true"` not a boolean `true`
wdt should we treat it as a string?

Reply:
disregard, it is desired with unescapeJson

Reply:
I think we need to be consistent. If we treat an object encoded in a string as a JSON object, I would expect similar behavior for boolean/array etc. What do you think?